FileDiscovery.py

```diff
@@ -0,0 +1,43 @@
+import glob
+import os
+
+
+class FileDiscovery:
+
+    def __init__(self, dir):
+        self.dir = dir
+        self.__extention_working()
+
+    def __extention_working(self):
+        # for number_files_in_dir
+        self.all_files_in_dir = os.listdir(self.dir)
+        self.path_to_all_files = [os.path.join(
+            self.dir, i) for i in self.all_files_in_dir]
+        # for all_extentions_in_dir
+        self.__splitted_files_list = set(
+            [i.split('.')[1] for i in self.all_files_in_dir])
+        # for unique_extention
+        extention_name = ['*.' + str(i) for i in self.__splitted_files_list]
```

Comment:
single responsibility principle
split this method into:
1. files discovery
2. file extension search